protocol/messenger_maps.go

```diff
@@ -53,10 +55,16 @@ func (cm *chatMap) Delete(chatID string) {
 */
 
 type contactMap struct {
-	sm sync.Map
+	sm     sync.Map
+	me     *Contact
+	logger *zap.Logger
 }
 
 func (cm *contactMap) Load(contactID string) (*Contact, bool) {
+	if contactID == cm.me.ID {
+		cm.logger.Warn("contacts map: loading own identity", zap.String("contactID", contactID))
```

Comment:
I wonder if `Debug` would be better. Is getting our own identity usually wrong?

Reply:
hmmm. Good question actually. We can check it together with https://github.com/status-im/status-go/issues/3719.

I would say that we shouldn't ask `contactsMap` for own identity and have this condition outside `Load` function. At the same time it's such a simple solution to use a cached built `Contact` internally, that I think we could keep it forever. If we stick to this, we can replace it with `Debug` or even remove at all.